src/Mod/Path/PathSimulator/App/VolSim.cpp

```diff
@@ -783,14 +783,14 @@ cSimTool::cSimTool(const TopoDS_Shape& toolShape, float res){
 
 float cSimTool::GetToolProfileAt(float pos)  // pos is -1..1 location along the radius of the tool (0 is center)
 {
-	try{
-		float radPos = std::abs(pos) * radius;
-		toolShapePoint test; test.radiusPos = radPos;
-		auto it = std::lower_bound(m_toolShape.begin(), m_toolShape.end(), test, toolShapePoint::less_than());
-		return it->heightPos;
-	}catch(...){
-		return 0;
-	}
+    if (m_toolShape.empty())
+        return 0.0f;
+
+    toolShapePoint test;
+    test.radiusPos = std::abs(pos) * radius;
+
+    auto it = std::lower_bound(m_toolShape.begin(), m_toolShape.end(), test, toolShapePoint::less_than());
+    return it->heightPos;
 }
```

Comment:
The cause of the crash was `it` being null and still being dereferenced. This is still a possibility if all values in `m_toolShape` are `< test`. See https://en.cppreference.com/w/cpp/algorithm/lower_bound. It appears they need `m_toolShape` to be sorted (or a looser condition defined in the link) before it can be run. If that is already satisfied, the following changes are good to go. If not, there may be a need to do the sorting here (which may be expensive) or elsewhere.

```suggestion
    toolShapePoint test;
    test.radiusPos = std::abs(pos) * radius;

    auto it = std::lower_bound(m_toolShape.begin(), m_toolShape.end(), test, toolShapePoint::less_than());
    
    if (it == m_toolShape.end())
        return 0.0;
    return it->heightPos;
}
```